### PR TITLE
Yash link files post

### PR DESCRIPTION
**Pull request by NeuralAlchemist**
- FE : able to post one image/ one video with text
- FE: able to show image or video inside postcard depending on what media a post contains
- FE : videos are playable
- BE : store videos in the file structure, store image as Byte array in database
- BE : add new mapping and handler to upload text with media

`UserStory 1`
As a pet owner, I want to create a post with images, so that I can share contents about my pet

`UserStory 2`
As a pet owner, I want to create a post with videos, so that I can share contents about my pet

### CODE REVIEW

File: src/main/java/se/kth/sda/skeleton/posts/Post.java [hunk 1]
```diff
@@ -42,11 +42,35 @@
...
+
+    private String videoName;
+
     // Constructor
     public Post() {
     }
 
+
     // Getters and Setters
+
```
Comment (eeels22) [inline src/main/java/se/kth/sda/skeleton/posts/Post.java]: perhaps we could remove this empty line

File: frontend/src/components/posts/PostForm.jsx [hunk 2]
```diff
@@ -1,26 +1,44 @@
 // NPM Packages
+
```
Comment (eeels22) [inline frontend/src/components/posts/PostForm.jsx]: pls remove empty line

File: frontend/src/components/posts/PostForm.jsx [hunk 3]
```diff
@@ -1,26 +1,44 @@
...
 import React, { useState } from 'react';
 import ResponsiveTextArea from '../ResponsiveTextArea';
+import PostsApi from "../../api/PostsApi";
 
-export default function PostForm({ onSubmit }) {
+export default function PostForm({ onSubmit, onSubmitMedia }) {
   // Local State
+
```
Comment (eeels22) [inline frontend/src/components/posts/PostForm.jsx]: pls remove empty line

File: frontend/src/components/posts/PostForm.jsx [hunk 4]
```diff
@@ -1,32 +1,48 @@
 // NPM Packages
+
 import React, { useState } from 'react';
 import ResponsiveTextArea from '../ResponsiveTextArea';
+import PostsApi from "../../api/PostsApi";
```
Comment (eeels22) [inline frontend/src/components/posts/PostForm.jsx:6]: Please add `// Project files` above the PostsApi import

File: frontend/src/components/posts/PostCard.jsx [hunk 5]
```diff
@@ -15,155 +16,166 @@ export default function PostCard({ post, currentUser, onDeleteClick }) {
     const [comments, setComments] = useState([]);
     const [toggleUpdatePost, setToggleUpdatePost] = useState(false);
```
Comment (eeels22) [inline frontend/src/components/posts/PostCard.jsx]: Would you mind reducing the indent on lines 15-17 to match the other lines?

**Review by eeels22 — APPROVED** [review]
Great job Yash! Just some really minor formatting things that could be tidied up if possible.

**Review by estherchee — APPROVED** [review]
Good job!

**Pull request merged**